testing/e2e/suites/nginxIngressController.go

```diff
@@ -112,6 +112,15 @@ func nicTests(in infra.Provisioned) []test {
 					return fmt.Errorf("able to create NginxIngressController despite missing Secret field'%s'", testNIC.Spec.ControllerNamePrefix)
 				}
 
+				testNIC = manifests.NewNginxIngressController("nginx-ingress-controller", "nginxingressclass")
+				testNIC.Spec.DefaultSSLCertificate = &v1alpha1.DefaultSSLCertificate{
+					KeyVaultURI: "Invalid.URI",
+				}
+				lgr.Info("creating NginxIngressController with invalid Secret field")
+				if err := c.Create(ctx, testNIC); err == nil {
+					return fmt.Errorf("able to create NginxIngressController despite invalid key vault uri'%s'", testNIC.Spec.ControllerNamePrefix)
+				}
+
```

Comment:
this e2e test only checks with an invalid keyvault uri. we need tests checking the positive scenario